hash_practice/exercises.py

```diff
@@ -22,8 +47,21 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)
+        Space Complexity: O(n)

```

Comment:
These would be accurate answers if we allowed arbitrary boards of nxn (well, actually the space complexity would be O(n^2) too), but since we only accept 3x3 boards, the time and space used will be constant and we can instead say it is O(1) in both time and space.